### PR TITLE
Add a helper function to delete etcd keys concurrently.

### PR DESCRIPTION
**Pull request by pphaneuf**

### CODE REVIEW

File: cpp/util/etcd_delete.cc [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+ private:
+  void RequestDone(Task* child_task);
+  void StartNextRequest(unique_lock<mutex>&& lock);
+
+  EtcdClient* const client_;
+  Task* const task_;
+  mutex mutex_;
+  int outstanding_;
+  int next_;
+  const vector<pair<string, int64_t> > keys_;
```
Comment (AlCutter) [inline cpp/util/etcd_delete.cc]: no need for the old school extra space in `> >` anymore :)

File: cpp/util/etcd_delete.cc [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    // In case the task uses an inline executor.
+    lock.unlock();
+    task_->Return(Status::CANCELLED);
+    return;
+  }
+
+  while (outstanding_ < FLAGS_etcd_delete_concurrency &&
+         next_ < keys_.size()) {
+    CHECK(lock.owns_lock());
+    const pair<string, int64_t>& key(keys_[next_]);
```
Comment (AlCutter) [inline cpp/util/etcd_delete.cc]: You could make `next_` a `vector<pair<string, int64_t>>::const_iterator` and then not need this line. (more for clarity if you think it'd help, I realise this is O(1) anyway.)

File: cpp/util/etcd_delete.cc [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    ++next_;
+    ++outstanding_;
+
+    // In case the task uses an inline executor.
+    lock.unlock();
```
Comment (AlCutter) [inline cpp/util/etcd_delete.cc]: case for a `ScopedUnlocker` ? :)

File: cpp/util/etcd_delete.cc (hunk 1 above)
Comment (pphaneuf) [inline cpp/util/etcd_delete.cc]: Done.

File: cpp/util/etcd_delete.cc (hunk 2 above)
Comment (pphaneuf) [inline cpp/util/etcd_delete.cc]: Can't really get away from it, but I replaced ```next_``` with ```it_```.

File: cpp/util/etcd_delete.cc (hunk 3 above)
Comment (pphaneuf) [inline cpp/util/etcd_delete.cc]: A weak one... I'll pass, if you'll allow? :smiley:

**Pull request merged**